project/scheduling/DecidableEquality.agda

```diff
@@ -0,0 +1,52 @@
+module DecidableEquality where
+open import Data.Nat using (ℕ)
+--import Data.Nat.Properties
+open import Relation.Binary.PropositionalEquality
+open ≡-Reasoning
+open import Data.Bool using (Bool ; true ; false ; if_then_else_ ; _∨_ ; _∧_)
+open import Data.List using (List ; [] ; _∷_ ; map ; all ; any ; and ; or )
+open import Data.List.Relation.Unary.AllPairs using (AllPairs; allPairs?)
+open import Data.List.Relation.Unary.All using (All)
+open import Data.Maybe using (Maybe ; nothing ; just)
+open import Data.Sum using (_⊎_)
+open import Level using (0ℓ)
+open import Relation.Unary using (Pred)
+open import Relation.Binary
+open import Relation.Nullary
+open import Relation.Nullary.Decidable using (True; False; ⌊_⌋; toWitness; fromWitness)
+open import Relation.Nullary.Negation
+open import Relation.Nullary.Product
+open import Relation.Nullary.Sum
+open import Data.Empty
+open import Data.Product using (_×_)
+open import Agda.Builtin.Unit
+
+record DecEq (A : Set) : Set where
```

Comment:
many of the functions in AcyclicGraph require Decidable Equality to be implemented for the datatypes it is indexed over. Record Instances make it alot nicer to pass around the instances of decidable equality for each data type less explicitly